articles/4_flex.md

```diff
@@ -0,0 +1,141 @@
+
+#### `flex-direction`
+
+This determines the layout's main axis, row or column. It also allows you to choose if you want the elements to move forwards or backwards. The options are `row`, `row-reverse`, `column`, `column-reverse`. The default value is `row`, meaning the elements flow from left to right.
+
+[Codepen Demo](http://codepen.io/raemadeline/pen/oLLawL)
+
+#### `flex-wrap`
+
+When you have too many elements and they all do not fit along the main axis in one line, you are given the choice as to whether they wrap around to a second line. By default, the value of `flex-wrap` is `nowrap`, so the elements will do their best to fit in the confined space. You can also set the value to `wrap`, which will start a second line after the first following the direction, and `wrap-reverse`, which will a second line before the first.
+
+[Codepen Demo](http://codepen.io/raemadeline/pen/jrreaQ)
+
+#### `flex-flow`
+
+This is just a shorthand to combine `flex-direction` and `flex-wrap`. The syntax is:
+
+```css
+.parent {
+  flex-flow: <'flex-direction'> || <'flex-wrap'>
+}
+```
+
+#### `justify-content`
+
+This determines the spacing of child elements along the main axis. This one is easier to understand visually, so here's an example:
+
+![Justify Content](../images/justify_content.png)
+
+To play around with these values, here's the [Codepen Demo](http://codepen.io/raemadeline/pen/BzzqxV). Try changing around other properties like `flex-direction` to see how these change.
+
+#### `align-items`
+
+`align-items` determines the spacing of child elements along the *secondary* axis, the one perpendicular to the axis specified in `flex-direction`.
+
+http://codepen.io/raemadeline/pen/WxGeNd
+
+#### `align-content`
+
+`align-content` is the same as `align-items`, but it handles the full range of child elements as a block instead of arranging each element in their line. For a better explanation, see the visual below (thanks css-tricks.com)
+
+![align-content](../images/align-content.png)
+
+### The Child Elements
+
+By applying the above properties to the parent element, you can create a lot of awesome layouts. However, you may occasionally want to customize a specific child element of the parent separately from the rest. These are properties you can add to the direct children of a flex parent.
+
+#### `order`
+
+This overrides the ordering of the elements as determined by the parent. By default, every child's `order` is determined by the layout. Generally, you should avoid this and just arrange your HTML such that the elements start in the correct order, but this is useful with responsive designs. See this [Codepen Demo](http://codepen.io/raemadeline/pen/PzGwGw) and try resizing the window. If you look at the HTML, the sidebar is before the main content, but its `order` gets overridden at smaller screen sizes to account for users looking at the page on mobile.
+
+#### `flex-grow`
```

Comment:
Actually, default isn't 1, items by default do not align themselves equally. You have to specifically set them else they just are as wide as their content. At some point it will try to distribute content around but it won't be as predictable without setting it. This is a good example of how the panel magic works, if you only set one element to 1, it will take up all the remaining space and the other elements will stay be as large as they need to be (I think this is much more obvious for a vertical layout)